extensions/import/src/wizard/pages/deployPlanPage.ts

```diff
@@ -0,0 +1,272 @@
+import * as nls from 'vscode-nls';
+import * as parser from 'htmlparser2';
+import { DacFxDataModel } from '../api/models';
+import { DataTierApplicationWizard } from '../dataTierApplicationWizard';
+import { DacFxConfigPage } from '../api/dacFxConfigPage';
+
+const localize = nls.loadMessageBundle();
+
+export enum reportSection {
+	Alert = 'Alert',
+	Operation = 'Operation',
+}
+
+export enum attributeName {
+	Name = 'Name',
+	Value = 'Value',
+	Type = 'Type',
+	Id = 'Id'
+}
+
+export class TableObject {
+	object: string;
+	type: string;
+	dataloss: boolean;
+}
+
+export class DeployPlanPage extends DacFxConfigPage {
+	protected readonly wizardPage: sqlops.window.modelviewdialog.WizardPage;
+	protected readonly instance: DataTierApplicationWizard;
+	protected readonly model: DacFxDataModel;
+	protected readonly view: sqlops.ModelView;
+	private formBuilder: sqlops.FormBuilder;
+	private form: sqlops.FormContainer;
+	private table: sqlops.TableComponent;
+	private loader: sqlops.LoadingComponent;
+	private dataLossCheckbox: sqlops.CheckBoxComponent;
+	private dataLossComponentGroup: sqlops.FormComponentGroup;
+	private noDataLossTextComponent: sqlops.FormComponent;
+
+	public constructor(instance: DataTierApplicationWizard, wizardPage: sqlops.window.modelviewdialog.WizardPage, model: DacFxDataModel, view: sqlops.ModelView) {
+		super(instance, wizardPage, model, view);
+	}
+
+	async start(): Promise<boolean> {
+		this.table = this.view.modelBuilder.table().component();
+		this.loader = this.view.modelBuilder.loadingComponent().withItem(this.table).component();
+		this.dataLossComponentGroup = await this.createDataLossComponents();
+		this.noDataLossTextComponent = await this.createNoDataLossText();
+
+		this.formBuilder = this.view.modelBuilder.formContainer()
+			.withFormItems(
+				[
+					{
+						component: this.loader,
+						title: ''
+					},
+					this.dataLossComponentGroup
+				], {
+					horizontal: true,
+				});
+		this.form = this.formBuilder.component();
+		await this.view.initializeModel(this.form);
+
+		return true;
+	}
+
+	async onPageEnter(): Promise<boolean> {
+		// reset checkbox settings
+		this.formBuilder.addFormItem(this.dataLossComponentGroup, { horizontal: true, componentWidth: 400 });
+		this.dataLossCheckbox.checked = false;
+		this.dataLossCheckbox.enabled = false;
+		this.formBuilder.removeFormItem(this.noDataLossTextComponent);
+
+		this.loader.loading = true;
+		this.table.data = [];
+		await this.populateTable();
+		this.loader.loading = false;
+		return true;
+	}
+
+	private async populateTable() {
+		let data = [];
+		let dataNoDataLoss = [];
+		let dataLossAlerts = new Set<string>();
+		let currentOperation = '';
+		let dataIssueAlert = false;
+		let currentReportSection: reportSection;
+		let currentTableObj: TableObject;
+
+		// parse the xml deploy plan
+		let report = await this.instance.generateDeployPlan();
+		let p = new parser.Parser({
+			onopentagname(name) {
+				if (name === 'Alert') {
```

Comment:
Can we create an DeploymentPlanXmlTags type with constants for 'Alert' and the  other xml tags?